internal/collector/processor/nodebatcher/node_batcher.go

```diff
@@ -0,0 +1,467 @@
+	return 0, nil
+}
+
+func (b *batcher) genBucketID(node *commonpb.Node, resource *resourcepb.Resource, spanFormat string) string {
+	h := md5.New()
+	if node != nil {
+		nodeKey, err := proto.Marshal(node)
+		if err != nil {
+			b.logger.Error("Error marshalling node to batcher mapkey.", zap.Error(err))
+		}
+		h.Write(nodeKey)
+	}
+	if resource != nil {
+		resourceKey, err := proto.Marshal(resource) // TODO: remove once resource is in span
+		if err != nil {
+			b.logger.Error("Error marshalling resource to batcher mapkey.", zap.Error(err))
+		}
+		h.Write(resourceKey)
+	}
+	return fmt.Sprintf("%x", h.Sum([]byte(spanFormat)))
+}
+
+func (b *batcher) getBucket(bucketID string) *nodeBatcher {
+	bucket, ok := b.buckets.Load(bucketID)
+	if ok {
+		return bucket.(*nodeBatcher)
+	}
+	return nil
+}
+
+func (b *batcher) getOrAddBucket(
+	bucketID string, node *commonpb.Node, resource *resourcepb.Resource, spanFormat string,
+) *nodeBatcher {
+	bucket, loaded := b.buckets.LoadOrStore(
+		bucketID,
+		newNodeBucket(b, node, resource, spanFormat, b.sendBatchSize, b.timeout, b.logger),
+	)
+	// Add this bucket to a random ticker
+	if !loaded {
+		stats.Record(context.Background(), statAddNodeBatches.M(1))
+		b.tickers[rand.Intn(len(b.tickers))].add(bucketID)
+	}
+	return bucket.(*nodeBatcher)
+}
+
+func (b *batcher) removeBucket(bucketID string) {
+	stats.Record(context.Background(), statRemoveNodeBatches.M(1))
+	b.buckets.Delete(bucketID)
+}
+
+type nodeBatcher struct {
+	timeout       time.Duration
+	sendBatchSize uint32
+	currBatch     *batch
+	node          *commonpb.Node
+	resource      *resourcepb.Resource // TODO(skaris) remove when resource added to span
+	spanFormat    string
+	lastSent      int64
+	logger        *zap.Logger
+
+	dead            uint32
+	cyclesUntouched uint32
+	parent          *batcher
+}
+
+func newNodeBucket(
+	parent *batcher,
+	node *commonpb.Node,
+	resource *resourcepb.Resource,
+	spanFormat string,
+	sendBatchSize uint32,
+	timeout time.Duration,
+	logger *zap.Logger,
+) *nodeBatcher {
+	nb := &nodeBatcher{
+		timeout:       timeout,
+		sendBatchSize: sendBatchSize,
+		currBatch:     newBatch(initBatchCap, sendBatchSize),
+		node:          node,
+		resource:      resource,
+		spanFormat:    spanFormat,
+		lastSent:      int64(0),
+		logger:        logger,
+
+		dead:            uint32(0),
+		cyclesUntouched: uint32(0),
+		parent:          parent,
+	}
+
+	return nb
+}
+
+func (nb *nodeBatcher) add(spans []*tracepb.Span) {
+	// Reset cycles untouched to 0
+	atomic.StoreUint32(&nb.cyclesUntouched, 0)
+	// Add will always add these spans to the current batch, if ok is false it means that
+	// we need to cut
+	var b *batch
+	closed := true
+	cut := false
+	for closed {
+		// atomic.LoadPointer only takes unsafe.Pointer interfaces. We do not use unsafe
+		// to skirt around the golang type system.
+		b = (*batch)(atomic.LoadPointer((*unsafe.Pointer)(unsafe.Pointer(&nb.currBatch))))
+		cut, closed = b.add(spans)
```

Comment:
So am assuming `cut` here means `performedBatching`? If so, please use a clearer term like `performedBatching`.

Reply:
used `cutBatch` instead (mirroring the function name) lmk what you think